src/lib/y2storage/encryption_processes/swap.rb

```diff
@@ -17,60 +17,169 @@
         enc
       end
 
-      private
-
-      # @see EncryptionProcesses::Base#encryption_type
+      # @see Base#encryption_type
       def encryption_type
         EncryptionType::PLAIN
       end
 
-      # Returns the options used for the encryption
-      #
-      # @return [Array<String>]
-      def crypt_options
-        [SWAP]
+      private
+
+      SWAP_OPTION = "swap".freeze
+      private_constant :SWAP_OPTION
+
+      class << self
+        private
+
+        # Encryption key file
+        #
+        # Each Swap process could use a different key file.
+        #
+        # @raise [RuntimeError] if no key file has been defined.
+        #
+        # @return [String]
+        def key_file
+          raise "No key file indicated!"
+        end
+
+        # Encryption cipher
+        #
+        # Each Swap process could use a different cipher.
+        #
+        # @return [String, nil] nil if no specific cipher
+        def cipher
+          nil
+        end
+
+        # Size of the encryption key
+        #
+        # @return [String, nil] nil if no specific size
+        def key_size
+          nil
+        end
+
+        # Encryption options to add to the encryption device
+        #
+        # @return [Array<String>]
+        def crypt_options
+          options = [swap_option]
+          options << cipher_option if cipher?
+          options << key_size_option if key_size?
+
+          options
+        end
+
+        # Wheter a specific cipher is used
+        #
+        # @return [Boolean]
+        def cipher?
+          !cipher.nil?
+        end
+
+        # Wheter a specific key size is used
+        #
+        # @return [Boolean]
+        def key_size?
+          !key_size.nil?
+        end
+
+        # Swap option for the crypttab file
+        #
+        # @return [String]
+        def swap_option
+          SWAP_OPTION
+        end
+
+        # Cipher option for the crypttab file
+        #
+        # @return [String, nil] nil if no specific cipher
+        def cipher_option
+          return nil unless cipher?
+
+          "cipher=#{cipher}"
+        end
+
+        # Key size option for the crypttab file
+        #
+        # @return [Boolean] nil if no specific size
+        def key_size_option
+          return nil unless key_size?
+
+          "size=#{key_size}"
+        end
+
+        # Checks whether the process was used according to the given key file and encrytion options
+        #
+        # @param key_file [String]
+        # @param crypt_options [Array<String>]
+        def used?(key_file, crypt_options)
+          contain_swap_option?(crypt_options) && use_key_file?(key_file)
+        end
+
+        # Whether the given encryption device contains the swap option
+        #
+        # @param crypt_options [Array<String>]
+        # @return [Boolean]
+        def contain_swap_option?(crypt_options)
+          crypt_options.any? { |o| o.downcase == swap_option }
```

Comment:
Just to be extra safe against future changes, we could use `o.downcase == swap_option.downcase`, or even better `swap_option.casecmp?(o)`